pkg/components/certpolicycontroller/v1/certpolicyctrl.go

```diff
@@ -60,21 +60,35 @@ func (addon AddonCertPolicyCtrl) GetManagedClusterAddOnName() string {
 
 // newCertPolicyControllerCR - create CR for component cert policy controller
 func newCertPolicyControllerCR(
-	instance *agentv1.KlusterletAddonConfig,
+	addonAgentConfig *agentv1.AddonAgentConfig,
 	namespace string,
 ) (*agentv1.CertPolicyController, error) {
 	labels := map[string]string{
-		"app": instance.Name,
+		"app": addonAgentConfig.ClusterName,
 	}
 
 	gv := agentv1.GlobalValues{
-		ImagePullPolicy: instance.Spec.ImagePullPolicy,
-		ImagePullSecret: instance.Spec.ImagePullSecret,
+		ImagePullPolicy: addonAgentConfig.ImagePullPolicy,
+		ImagePullSecret: addonAgentConfig.ImagePullSecret,
 		ImageOverrides:  make(map[string]string, 1),
-		NodeSelector:    instance.Spec.NodeSelector,
+		NodeSelector:    addonAgentConfig.NodeSelector,
+	}
+	switch addonAgentConfig.KlusterletAddonConfig.Spec.CertPolicyControllerConfig.ProxyPolicy {
+	case agentv1.ProxyPolicyOCPGlobalProxy:
+		gv.ProxyConfig = map[string]string{
+			agentv1.HTTPProxy:  addonAgentConfig.KlusterletAddonConfig.Status.OCPGlobalProxy.HTTPProxy,
+			agentv1.HTTPSProxy: addonAgentConfig.KlusterletAddonConfig.Status.OCPGlobalProxy.HTTPSProxy,
+			agentv1.NoProxy:    addonAgentConfig.KlusterletAddonConfig.Status.OCPGlobalProxy.NoProxy,
```

Comment:
should user be able to set no proxy from spec?

Reply:
if the proxyPolicy is ocpGlobalporxy, will set no proxy from status of addonConfig.
if the proxyPolicy is customProxy, will set no proxy from spec of addonConfig.